`make dev` fails on macOS because "detected dubious ownership in repository"

## Description

On macOS, `make dev` fails in—

https://github.com/freedomofpress/securedrop/blob/50e89ad43c8124e6ea411d53671bf7af9bc4cf1c/securedrop/bin/run#L6-L8

—with the error:

```sh-session
cfm@ozymandias securedrop % make dev
[...]
************************************************************
Exposed services will be available on localhost at
Source interface: http://127.0.0.1:8080
Journalist interface: http://127.0.0.1:8081
************************************************************
fatal: detected dubious ownership in repository at '/Users/cfm/proj/fpf/securedrop'
To add an exception for this directory, call:

        git config --global --add safe.directory /Users/cfm/proj/fpf/securedrop
/Users/cfm/proj/fpf/securedrop/securedrop/bin/run: line 8: cd: /securedrop: No such file or directory
make: *** [dev] Error 1
```

## Steps to Reproduce

* macOS 12.6.1 (21G217)
* `git version 2.37.1 (Apple Git-137.1)`

```sh-session
cfm@ozymandias securedrop % make dev
```

## Expected Behavior

The development environment starts.

## Actual Behavior

```sh-session
[...]
************************************************************
Exposed services will be available on localhost at
Source interface: http://127.0.0.1:8080
Journalist interface: http://127.0.0.1:8081
************************************************************
fatal: detected dubious ownership in repository at '/Users/cfm/proj/fpf/securedrop'
To add an exception for this directory, call:

        git config --global --add safe.directory /Users/cfm/proj/fpf/securedrop
/Users/cfm/proj/fpf/securedrop/securedrop/bin/run: line 8: cd: /securedrop: No such file or directory
make: *** [dev] Error 1
```
## Comments

I can work around this locally with:

```patch
--- a/securedrop/bin/run
+++ b/securedrop/bin/run
@@ -3,6 +3,7 @@
 
 set -eu
 
+git config --global --add safe.directory /Users/cfm/proj/fpf/securedrop
```

But we should:
1. [x] see if @eaon and I can reproduce this tomorrow;
2. [x] placate Git's permission checks generalizably; and
3. [x] make `run` error out when `$REPOROOT` is unset.

--- a/securedrop/bin/run
+++ b/securedrop/bin/run
@@ -3,7 +3,9 @@
 
 set -eu
 
-export REPOROOT="${REPOROOT:-$(git rev-parse --show-toplevel)}"
+cd ..
+export REPOROOT="${REPOROOT:-$PWD}"
+git config --global --add safe.directory "$REPOROOT"
 
 cd "${REPOROOT}/securedrop"
 

